### PR TITLE
Fix Rdatasets#each with mixed data of numeric and string

### PR DESCRIPTION
**Pull request by heronshoes**
This pull request is a solution of 2nd part in https://github.com/red-data-tools/red-datasets/issues/138 .

This code is based on #139 and at 1 commit ahead. 
If #139 merged, I will rebase this on master.

### CODE REVIEW

**Comment by kou** [comment]
We may be able to pass "quoted" information to converters via `CSV::FieldInfo`: https://github.com/ruby/csv/blob/master/lib/csv/fields_converter.rb#L66

**Comment by heronshoes** [comment]
Thank you for your advice. I agree with you that this code is redundant. I will try to use FieldInfo.

I will appreciate it if you could continue to support.

**Comment by heronshoes** [comment]
This change will cover both "germination" and "nace_rev2". I added tests for them.
For the "nace_rev2" is temporally one. I think we should some refinement in CSV library itself.

File: lib/datasets/rdatasets.rb [hunk 1]
```diff
@@ -82,23 +82,52 @@ def each(&block)
...
+      quote_preserving_converter = lambda do |field, field_info|
+        f = field.encode(CSV::ConverterEncoding)
+
+        # temporally code for "validate/nace_rev2"
+        if [@package_name, @dataset_name] == %w[validate nace_rev2]
+          quoted = {Level: false, Order: false}
+          return f if quoted.fetch(field_info.header, true)
         end
+
+        m = /"(.+)"/.match(f)
```
Comment (kou) [inline lib/datasets/rdatasets.rb:113]:
```suggestion
        m = /\A"(.+)"\z/.match(f)
```

The following code may be faster than it:

```ruby
  return f[1..-2] if f.start_with?('"') and f.end_with?('"')
```

File: lib/datasets/rdatasets.rb [hunk 2]
```diff
@@ -82,23 +82,52 @@ def each(&block)
...
+        return m[1] if m
+        return DateTime.parse(f) if f.match?(DateTimeMatcher) rescue
+        return Integer(f) rescue
+        return Float(f) rescue field
```
Comment (kou) [inline lib/datasets/rdatasets.rb:117]: How about splitting the default case for consistency?

```ruby
  return Float(f) rescue
  field
```

**Comment by heronshoes** [comment]
I want to hold this until 'quoted?' information is available from CSV. I memorize the fixes needed.

- Use start_with?/end_with? instead of regexp.
- Use begin/rescue for readability.
- Will not use :quote_char option when CSV improved.
- Use :symbol_raw header_converter in CSV.

**Comment by heronshoes** [comment]
Sorry for holding so long !

I reflected CSV's 'quoted' feature and fixed remaining issues.

Temporally I specify HEAD version of CSV to use the new feature.
```
# in Gemfile
gem 'csv', github: 'ruby/csv'
```

I would appreciate it if you could comment.

File: Gemfile [hunk 3]
```diff
@@ -3,3 +3,5 @@
 source "https://rubygems.org/"
 
 gemspec
+
+gem 'csv', github: 'ruby/csv'
```
Comment (kou) [inline Gemfile:7]: I've released csv gem. Could you remove this and updated required csv version to 3.2.4?

Comment (heronshoes) [reply Gemfile:7]: OK 👍 Thanks!

Comment (heronshoes) [reply Gemfile]: I appreciate if you could see it.

**Comment by kou** [comment]
Thanks!

**Pull request merged**